docs/proposals/failover/application-failover.md

```diff
@@ -0,0 +1,566 @@
+At this time, users want to schedule the application to a cluster other than the one that failed previously.
+
+Link:
+
+* en: [Use preemptible instances(Alibaba Cloud)](https://www.alibabacloud.com/help/en/container-service-for-kubernetes/latest/use-preemptible-instances), [Spot Mode(Tencent Cloud)](https://www.tencentcloud.com/document/product/457/46967), [Amazon EC2 Spot(AWS)](https://aws.amazon.com/ec2/spot/details/)
+* cn: [使用抢占式实例（阿里云）](https://help.aliyun.com/document_detail/165053.html), [竞价模式说明（腾讯云）](https://cloud.tencent.com/document/product/457/59364), [Amazon EC2 Spot功能（AWS）](https://aws.amazon.com/cn/ec2/spot/details/?nc1=h_ls)
+
+### Dependency
+
+Karmada's `Resource Interpreter Framework` is designed for interpreting resource structure. It provides users with a interpreter operation to tell Karmada
+how to figure out the health status of a specific object. It is up to users to decide when to reschedule.
+
+For example:
+
+```yaml
+apiVersion: config.karmada.io/v1alpha1
+kind: ResourceInterpreterCustomization
+metadata:
+  name: declarative-configuration-example
+spec:
+  target:
+    apiVersion: apps/v1
+    kind: Deployment
+  healthInterpretation:
+    luaScript: >
+      function InterpretHealth(observedObj)
+        return observedObj.status.readyReplicas == observedObj.spec.replicas
+      end
+```
+
+### Risks and Mitigations
+
+This proposal maintains the backward compatibility, the system built with previous versions of Karmada can be seamlessly migrated to the new version.
+The previous configurations(yamls) could be applied to the new version of Karmada and without any behavior change.
+
+## Design Details
+
+Users are able to define the behavior of failover by defining policies.
+The configuration of failover behavior includes:
+* When to trigger application failover
+* How to migrate from one cluster to another
+* Time at which the migrated cluster can be recovered
+
+Sometimes, you have to deal with legacy applications that might require an additional startup time on their first initialization. Or you want to 
+avoid invalid failovers due to misconfigurations. In such cases, we need a parameter to init the failover behavior.
+Optional configuration items include:
+* Healthy State. We just want to migrate those applications that have run successfully. For those applications that failed at the beginning, we prefer to solve them in the single cluster.
+* A Period of time. We can wait a period of time before the application failover is initiated.
+
+For some short-lived exceptions(which can be recovered after a period of time), we don't want to immediately migrate the application to other clusters which may make the migration very frequent.
+Therefore, we need a parameter to tolerate the unhealthy application.
+
+### API change
+
+#### PropagationPolicy API change
+
+Option One:
```

Comment:
We don't need to highlight this is `option one`.  The design present here means this is the proposing option.